db/schema.rb

```diff
@@ -0,0 +1,86 @@
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 20170304172631) do
+
+  create_table "compositions", force: :cascade do |t|
+    t.string   "name"
+    t.text     "notes"
+    t.integer  "map_id",     null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["map_id"], name: "index_compositions_on_map_id"
+  end
+
+  create_table "hero_pools", force: :cascade do |t|
+    t.integer  "player_id",              null: false
+    t.integer  "hero_id",                null: false
+    t.integer  "confidence", default: 0, null: false
+    t.datetime "created_at",             null: false
+    t.datetime "updated_at",             null: false
+    t.index ["hero_id"], name: "index_hero_pools_on_hero_id"
+    t.index ["player_id"], name: "index_hero_pools_on_player_id"
+  end
+
+  create_table "heroes", force: :cascade do |t|
+    t.string   "name",       null: false
+    t.string   "role"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "maps", force: :cascade do |t|
+    t.string   "name",       null: false
+    t.string   "type"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "player_selections", force: :cascade do |t|
+    t.integer  "player_id",      null: false
+    t.integer  "hero_id",        null: false
+    t.string   "role"
+    t.integer  "composition_id", null: false
+    t.datetime "created_at",     null: false
+    t.datetime "updated_at",     null: false
+    t.index ["composition_id"], name: "index_player_selections_on_composition_id"
+    t.index ["hero_id"], name: "index_player_selections_on_hero_id"
+    t.index ["player_id"], name: "index_player_selections_on_player_id"
+  end
+
+  create_table "players", force: :cascade do |t|
+    t.string   "name",       null: false
+    t.string   "battlenet"
+    t.integer  "user_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["user_id"], name: "index_players_on_user_id"
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.string   "email",                  default: "", null: false
+    t.string   "encrypted_password",     default: "", null: false
+    t.string   "reset_password_token"
+    t.datetime "reset_password_sent_at"
+    t.datetime "remember_created_at"
+    t.integer  "sign_in_count",          default: 0,  null: false
+    t.datetime "current_sign_in_at"
+    t.datetime "last_sign_in_at"
+    t.string   "current_sign_in_ip"
+    t.string   "last_sign_in_ip"
+    t.datetime "created_at",                          null: false
+    t.datetime "updated_at",                          null: false
+    t.index ["email"], name: "index_users_on_email", unique: true
+    t.index ["reset_password_token"], name: "index_users_on_reset_password_token", unique: true
+  end
```

Comment:
I saw the discussion of player separate from user in the original issue, but I'm not sure we gain much from that separation. If we treat every user as a player, then we can provide freedom in our business logic from needing to branch based on the user type. What I mean is, building a composition that doesn't include yourself, should be a valid composition, but I don't think that requires us to have a separate user model. That feels like a business logic concern rather than a data model concern. I think having two separate user types here will make our business logic unnecessarily more complex starting out.